pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1986,52 +1986,90 @@ cdef class YearBegin(YearOffset):
 # Quarter-Based Offset Classes
 
 cdef class QuarterOffset(SingleConstructorOffset):
-    _attributes = tuple(["n", "normalize", "startingMonth"])
+    """
+    Base for quarter-based offset classes.
+
+    Parameters
+    ----------
+    n : int, default 1
+        Number of quarters to offset.
+    normalize : bool, default False
+        If true, the time component of the resulting date-time is converted
+        to 00:00:00, i.e. midnight (the start, not the end of date-time).
+    month : int
+        The calendar month number (1 for January) of the beginning or ending
+        month in a custom-defined quarter to be used as an offset.
+
+        .. versionadded:: 1.X.X
+    startingMonth : int
+        The calendar month number (1 for January) of the beginning or ending
+        month in a custom-defined quarter to be used as an offset.
+
+        .. deprecated:: 1.X.X
+    """
+
+    _attributes = tuple(["n", "normalize", "month"])
     # TODO: Consider combining QuarterOffset and YearOffset __init__ at some
-    #       point.  Also apply_index, is_on_offset, rule_code if
-    #       startingMonth vs month attr names are resolved
+    #       point. Also apply_index, is_on_offset, rule_code.
 
-    # FIXME: python annotations here breaks things
-    # _default_starting_month: int
-    # _from_name_starting_month: int
+    # FIXME: python annotation here breaks things
 
     cdef readonly:
-        int startingMonth
+        int month
+        int startingMonth  # Backwards compatibility
 
-    def __init__(self, n=1, normalize=False, startingMonth=None):
+    def __init__(self, n=1, normalize=False, month=None,
+                 *, startingMonth=None):  # Backwards compatibility
         BaseOffset.__init__(self, n, normalize)
 
-        if startingMonth is None:
-            startingMonth = self._default_starting_month
-        self.startingMonth = startingMonth
+        # Backwards compatibility
+        if month and startingMonth and month != startingMonth:
+            raise TypeError("Offset received both month and startingMonth")
```

Comment:
is this line tested?

Reply:
Thank you, I missed that one. Now it is tested.